partiql-lang/src/main/kotlin/org/partiql/lang/eval/ExprFunction.kt

```diff
@@ -48,39 +58,16 @@ interface ExprFunction {
      * @param required The required arguments.
      */
     fun callWithRequired(session: EvaluationSession, required: List<ExprValue>): ExprValue {
-        // Deriving ExprFunctions must implement this if they have a valid call form including only required parameters
         errNoContext("Invalid implementation for ${signature.name}#call", ErrorCode.INTERNAL_ERROR, true)
     }
 
-    /**
-     * Invokes the function with its required parameters and an optional parameter.
-     *
-     * It is assumed that the [ExprFunction]s will always be called after validating
-     * the arguments for the `arity` and the types of arguments.
-     * [EvaluatingCompiler] validates the arguments before calling [ExprFunction]s.
-     * Hence the implementations are not required to deal with it.
-     *
-     * @param session The current [EvaluationSession].
-     * @param required The required arguments.
-     * @param opt The optional arguments.
-     */
+    @Deprecated("As function overloading is now supported, please provide multiple implementations of ExprFunction and invoke callWithRequired() instead.", level = DeprecationLevel.ERROR)
```

Comment:
```suggestion
    @Deprecated("Please define overloaded functions by providing each alternative as its own ExprFunction; each function is invoked by callWithRequired() rather than callWithOptional().", level = DeprecationLevel.ERROR)
```